kyo-core/shared/src/main/scala/kyo/Hub.scala

```diff
@@ -74,74 +83,108 @@ class Hub[A] private[kyo] (
       */
     def closed(using Frame): Boolean < IO = ch.closed
 
-    /** Closes the Hub and all its listeners.
+    /** Closes the Hub and returns any remaining messages.
+      *
+      * When closed:
+      *   - The Hub stops accepting new messages
+      *   - All listeners are automatically closed
+      *   - Any blocked publishers are unblocked with a Closed failure
+      *   - All subsequent operations will fail with Closed
       *
       * @return
-      *   a Maybe containing any remaining elements in the Hub
+      *   a Maybe containing any messages that were in the Hub's buffer at the time of closing. Returns Absent if the close operation fails
+      *   (e.g., if the Hub was already closed).
       */
     def close(using frame: Frame): Maybe[Seq[A]] < IO =
-        fiber.interruptDiscard(Result.Panic(Closed("Hub", initFrame))).andThen {
+        fiber.interruptDiscard(Result.Failure(Closed("Hub", initFrame))).andThen {
             ch.close.map { r =>
                 IO {
-                    val array = listeners.toArray()
-                    discard(listeners.removeIf(_ => true))
-                    Loop.indexed { idx =>
-                        if idx == array.length then Loop.done
-                        else
-                            array(idx).asInstanceOf[Channel[A]].close
-                                .map(_ => Loop.continue)
-                    }.andThen(r)
+                    val l = Chunk.from(listeners.toArray()).asInstanceOf[Chunk[Listener[A]]]
+                    discard(listeners.clear())
+                    Kyo.foreachDiscard(l)(_.child.close.unit).andThen(r)
                 }
             }
         }
 
-    /** Creates a new listener for this Hub with default buffer size.
+    /** Creates a new listener for this Hub with the default buffer size.
       *
       * @return
-      *   a new Listener
+      *   a new Listener that will receive messages from the Hub
       */
     def listen(using Frame): Listener[A] < (IO & Abort[Closed] & Resource) =
-        listen(0)
+        listen(DefaultBufferSize)
```

Comment:
Use a default buffer size of `4096` instead of `0`. Hub's approach with buffers for listeners is designed to reduce contention in high-throughput scenarios. A default of `0` goes against this design.